src/Contracts/Repository.php

```diff
@@ -64,5 +64,5 @@ public function getChangedAttributes(DomainEntity $entity): array;
 
     public function getChangedAttribute(DomainEntity $entity, string $name);
 
-    public function wasAttributeChanged(DomainEntity $entity, string $name): bool;
+    public function wasAttributeChanged(DomainEntity $entity, string $name, bool $checkByValue = false): bool;
```

Comment:
I don't think it's good to add a flag for such a task. I believe it's better to add a new method called `wasAttributeValueChanged` and mark the current one as deprecated due to the typecast issue inside of Yii itself.